### DISCUSSION TITLE
iOS automation in windows

### DISCUSSION THREAD

**Discussion by mega6453** [danielpaulus/go-ios#256] [Q&A]
Hi @danielpaulus 
I've tried using **"ios-appium-on-linux"** in linux and I can able to run iOS automation. But how to do the same in windows OS? I've downloaded the go-ios windows exe file. I've downloaded libimobiledevice windows executables files which has the usbmuxd and all.  I tried few ways but nothing didn't work. 
Forwarding the port 7777 to 8100 using go-ios.
Started appium from node
Did ios runwda - launched the wda in phone.

Not sure how to make the usbmuxd to work in windows...

Appium throws the following error. Can u help me please?

[debug] [BaseDriver] Event 'xcodeDetailsRetrieved' logged at 1683711446820 (15:07:26 GMT+0530 (India Standard Time))
[debug] [BaseDriver] Event 'appConfigured' logged at 1683711446821 (15:07:26 GMT+0530 (India Standard Time))
[debug] [BaseDriver] Event 'resetStarted' logged at 1683711446822 (15:07:26 GMT+0530 (India Standard Time))
[debug] [XCUITest] Reset: running ios real device reset flow
[debug] [BaseDriver] Event 'resetComplete' logged at 1683711446823 (15:07:26 GMT+0530 (India Standard Time))
[WebDriverAgent] Using WDA path: 'C:\Users\mc\AppData\Roaming\npm\node_modules\appium\node_modules\appium-webdriveragent'
[WebDriverAgent] Using WDA agent: 'C:\Users\mc\AppData\Roaming\npm\node_modules\appium\node_modules\appium-webdriveragent\WebDriverAgent.xcodeproj'
[XCUITest] {"code":"ERR_INVALID_ARG_TYPE"}
[DevCon Factory] Releasing connections for udid-xxxxxxxxxx-xxxxxxxxxxx-udid device on any port number
[DevCon Factory] No cached connections have been found
[WebDriverAgent] Cannot retrieve WDA build settings. Original error: spawn xcodebuild ENOENT
[debug] [BaseDriver] Event 'newSessionStarted' logged at 1683711446844 (15:07:26 GMT+0530 (India Standard Time))
[debug] [W3C] Encountered internal error running command: TypeError: The "path" argument must be of type string. Received undefined

**Comment by danielpaulus** [comment]
Interesting that you ask this now. Maybe join the Discord community, we are actually right now discussing bundling the iOS driver and usbmux with go-ios for a more straightforward experience on Windows. 
If you can, join us and share your use case with us there so we can figure out how to best solve it 
https://discord.gg/Zr8J3bCdkv

**Reply by danielpaulus** [reply]
Oh and also there seems to be a bug.

**Reply by mega6453** [reply]
Joined daniel. Thanks for inviting.

**Reply by mega6453** [reply]
@danielpaulus You have mentioned "there seems to be a bug". May I know which one you are referring to? In go-ios or usbmuxd windows executable?
